<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CanOperateDialog</class>
 <widget class="QDialog" name="CanOperateDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>914</width>
    <height>197</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CAN操作</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="baseTab">
      <attribute name="title">
       <string>基本操作</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="sendFrameGroupBox">
         <property name="title">
          <string>发送数据帧</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QComboBox" name="chanelComboBox">
              <item>
               <property name="text">
                <string>通道 0</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>通道 1</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="sendTypeLabel">
              <property name="maximumSize">
               <size>
                <width>60</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>发送方式：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="sendTypeComboBox">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <item>
               <property name="text">
                <string>正常发送</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>单次发送</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>自发自收</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>单次自发自收</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="frameTypeLabel">
              <property name="text">
               <string> 帧类型：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="frameTypeComboBox">
              <property name="minimumSize">
               <size>
                <width>80</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>80</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>标准帧</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>扩展帧</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="frameFormatLabel">
              <property name="text">
               <string>帧格式：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="frameFormatComboBox">
              <property name="minimumSize">
               <size>
                <width>80</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>80</width>
                <height>16777215</height>
               </size>
              </property>
              <item>
               <property name="text">
                <string>数据帧</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>远程帧</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="eachFrameNumLabel">
              <property name="text">
               <string>  每次帧数：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="sendFrameNumSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="sendIntervalMsLabel">
              <property name="text">
               <string>  发送间隔(ms)：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="sendIntervalMsSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>300000</number>
              </property>
              <property name="singleStep">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLabel" name="frameIDLabel">
              <property name="text">
               <string>帧ID：0x</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="frameIDLineEdit">
              <property name="maximumSize">
               <size>
                <width>80</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>00000080</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="DLCLabel">
              <property name="text">
               <string>DLC：0x</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="DLCLineEdit">
              <property name="minimumSize">
               <size>
                <width>30</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>08</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="sendDataLabel">
              <property name="maximumSize">
               <size>
                <width>60</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>数据(Hex)：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="sendDataLineEdit">
              <property name="text">
               <string>01 02 03 04 05 06 07 08</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="sendDataButton">
              <property name="text">
               <string>发送</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="periodTab">
      <attribute name="title">
       <string> 周期操作</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QListWidget" name="periodListWidget"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="playTab">
      <attribute name="title">
       <string> 播放操作</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>类型：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="playTypeComboBox">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>顺序播放</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>时间播放</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>循环播放</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>每次帧数：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="frameNumSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string> 每次发送时间间隔(ms)：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="timeIntervalSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>300000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="filenameLabel">
           <property name="text">
            <string>文件名：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="filepathLineEdit"/>
         </item>
         <item>
          <widget class="QProgressBar" name="filepathProgressBar">
           <property name="value">
            <number>24</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="openFileToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="playButton">
           <property name="text">
            <string>播放</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
